src/atc/delta/deltastream_handle.py

```diff
@@ -0,0 +1,239 @@
+from typing import List
+
+from pyspark.sql import DataFrame
+from pyspark.sql.streaming import DataStreamWriter
+
+from atc.configurator.configurator import Configurator
+from atc.functions import init_dbutils
+from atc.spark import Spark
+from atc.tables.SparkHandle import SparkHandle
+from atc.utils import GetMergeStatement
+from atc.utils.FileExists import file_exists
+
+
+class DeltaStreamHandle(SparkHandle):
+    def __init__(
+        self,
+        *,
+        checkpoint_path: str,
+        name: str = None,
+        data_format: str = None,
+        location: str = None,
+        trigger_type: str = "availablenow",
+        trigger_time: str = None,
+        await_termination=False,
+    ):
+        """
+        name: name of the delta table
+
+        checkpoint_path: The location of the checkpoints, <table_name>/_checkpoints
+            The Delta Lake VACUUM function removes all files not managed by Delta Lake
+            but skips any directories that begin with _. You can safely store
+            checkpoints alongside other data and metadata for a Delta table
+            using a directory structure such as <table_name>/_checkpoints
+            See: https://docs.databricks.com/structured-streaming/delta-lake.html
+
+        location: the location of the delta table (Optional)
+
+        data_format: the data format of the files that are read (Default delta)
+
+        trigger_type: the trigger type of the stream.
+            See: https://docs.databricks.com/structured-streaming/triggers.html
+
+        trigger_time: if the trigger has is "processingtime",
+            it should have a trigger time associated
+
+        awaitTermination: if true, the ETL will wait for the termination of THIS query.
+
+        """
+        assert (
+            Spark.version() >= Spark.DATABRICKS_RUNTIME_10_4
+        ), f"DeltaStreamHandle not available for Spark version {Spark.version()}"
+
+        super().__init__(name, location, data_format)
+
+        self._checkpoint_path = checkpoint_path
+        self._trigger_type = trigger_type.lower() if trigger_type else "availablenow"
+        self._trigger_time = trigger_time.lower() if trigger_time else None
+        self._awaitTermination = await_termination if await_termination else True
```

Comment:
Does not correspons to default value in __ini__

Reply:
maybe it shouldnt be the configurator that should determine this... It should perhaps be a input?

Reply:
@JeppeBlixen please add your view on this.